File: python-sdk/aquin/exceptions.py
class AquinError(Exception):
    """Base exception for all Aquin SDK errors."""
    def __init__(self, message: str, status_code: int = None):
        super().__init__(message)
        self.status_code = status_code


class InvalidKeyError(AquinError):
    """API key is invalid or disabled."""
    pass


class InsufficientCreditsError(AquinError):
    """Account does not have enough credits."""
    pass


class RateLimitError(AquinError):
    """Too many requests."""
    pass


class ModelNotFoundError(AquinError):
    """The model for this API key has not been trained yet."""
    pass


class InferenceError(AquinError):
    """The inference server failed."""
    pass

File: javascript-sdk/src/exceptions.ts
export class AquinError extends Error {
  statusCode?: number;
  constructor(message: string, statusCode?: number) {
    super(message);
    this.name = "AquinError";
    this.statusCode = statusCode;
  }
}

export class InvalidKeyError extends AquinError {
  constructor(message: string) {
    super(message, 401);
    this.name = "InvalidKeyError";
  }
}

export class InsufficientCreditsError extends AquinError {
  constructor(message: string) {
    super(message, 402);
    this.name = "InsufficientCreditsError";
  }
}

export class RateLimitError extends AquinError {
  constructor(message: string) {
    super(message, 429);
    this.name = "RateLimitError";
  }
}

export class ModelNotFoundError extends AquinError {
  constructor(message: string) {
    super(message, 404);
    this.name = "ModelNotFoundError";
  }
}

export class InferenceError extends AquinError {
  constructor(message: string, statusCode?: number) {
    super(message, statusCode);
    this.name = "InferenceError";
  }
}